leakcanary-android/src/main/java/com/squareup/leakcanary/internal/LeakCanarySingleThreadFactory.java

```diff
@@ -0,0 +1,30 @@
+package com.squareup.leakcanary.internal;
+
+import java.util.concurrent.ThreadFactory;
+
+/**
+ * Copied from {@link java.util.concurrent.Executors#defaultThreadFactory()}. This is intended to
+ * only be used with a single thread executor.
+ */
+final class LeakCanarySingleThreadFactory implements ThreadFactory {
```

Comment:
This does way more than what we need (ie we won't need to deal with thread groups or change `daemon` or `priority` or access the security manager).

A simpler implementation would do well:

```java
final class LeakCanaryThreadFactory implements ThreadFactory {
private final String threadName;

LeakCanaryThreadFactory(String threadName) {
  this.threadName = "LeakCanary-" + threadName;
}

  @Override public Thread newThread(Runnable runnable) {
    return new Thread(runnable, threadName);
  }
}
```